### DISCUSSION TITLE
TRUSTEE_W doesn't have its inner union types

### DISCUSSION THREAD

**Discussion by 0x5bfa** [microsoft/CsWin32#1232] [Q&A] [answered]
The definition of it is below:

```c
typedef struct _TRUSTEE_W {
  struct _TRUSTEE_W          *pMultipleTrustee;
  MULTIPLE_TRUSTEE_OPERATION MultipleTrusteeOperation;
  TRUSTEE_FORM               TrusteeForm;
  TRUSTEE_TYPE               TrusteeType;
  union {
    LPWSTR             ptstrName;
    SID                *pSid;
    OBJECTS_AND_SID    *pObjectsAndSid;
    OBJECTS_AND_NAME_W *pObjectsAndName;
  };
  LPWCH                      ptstrName;
} TRUSTEE_W, *PTRUSTEE_W, TRUSTEEW, *PTRUSTEEW;
```

However, in CsWin32, it produces as below (removed comments):
```cs
[global::System.CodeDom.Compiler.GeneratedCode("Microsoft.Windows.CsWin32", "0.3.106+a37a0b4b70")]
public partial struct TRUSTEE_W
{
    public unsafe winmdroot.Security.Authorization.TRUSTEE_W* pMultipleTrustee;
    public winmdroot.Security.Authorization.MULTIPLE_TRUSTEE_OPERATION MultipleTrusteeOperation;
    public winmdroot.Security.Authorization.TRUSTEE_FORM TrusteeForm;
    public winmdroot.Security.Authorization.TRUSTEE_TYPE TrusteeType;
    public winmdroot.Foundation.PWSTR ptstrName;
}
```

If I'm wrong please correct me.

Thanks!

**Comment by 0x5bfa** [comment]
Btw, how can I find [`ACCESS_MASK`](https://learn.microsoft.com/en-us/windows/win32/secauthz/access-mask)?

![image](https://github.com/microsoft/CsWin32/assets/62196528/66ea83db-9793-4cd7-a401-06082418903a)

**Reply by AArnott** [reply]
There isn't an `ACCESS_MASK` defined in the metadata. The constants you're looking for are kind of spread out. See https://github.com/microsoft/win32metadata/issues/1948 for details.

**Comment by AArnott** [comment, marked as answer]
CsWin32 doesn't represent the unions because the metadata doesn't. But this looks like a bug to me. Can you file one over at https://github.com/microsoft/win32metadata/issues?

**Reply by 0x5bfa** [reply]
Sure! I suppose they should have Anonymous property.
